### PR TITLE
config_consistency: test ClientIPHeader

### PR DESCRIPTION
**Pull request by mabdinur**
## Description

- Introduces a test that validates that the `DD_TRACE_CLIENT_IP_HEADER` configuration is supported across client libraries
- Enables this test for the following libraries:  python, ... 

## Workflow

1. ⚠️ Create your PR as draft ⚠️
2. Work on you PR until the CI passes (if something not related to your task is failing, you can ignore it)
3. Mark it as ready for review
    * Test logic is modified? -> Get a review from RFC owner. We're working on refining the `codeowners` file quickly.
    * Framework is modified, or non obvious usage of it -> get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

:rocket: Once your PR is reviewed, you can merge it!

🛟 [#apm-shared-testing](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X) 🛟

## Reviewer checklist

* [ ] If PR title starts with `[<language>]`, double-check that only `<language>` is impacted by the change
* [ ] No system-tests internal is modified. Otherwise, I have the approval from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)
* [ ] CI is green, or failing jobs are not related to this change (and you are 100% sure about this statement)
* [ ] A docker base image is modified?
    * [ ] the relevant `build-XXX-image` label is present
* [ ] A scenario is added (or removed)?
    * [ ] Get a review from [R&P team](https://dd.enterprise.slack.com/archives/C025TJ4RZ8X)

### CODE REVIEW

File: tests/test_config_consistency.py [hunk 1]
```diff
@@ -178,33 +178,46 @@ def test_query_string_redaction(self):
         assert _get_span_by_tags(trace, expected_tags), f"Span with tags {expected_tags} not found in {trace}"
 
 
-def _get_span(spans, resource_name, tags):
```
Comment (mabdinur) [inline tests/test_config_consistency.py:181]: @zacharycmontoya I replaced `_get_span` with `_get_span_by_tags`. I found that this approach works best with most libraries. Resource names are not set in a consistent way across client libraries. This is something we can work with IDM to address.

File: utils/_context/_scenarios/__init__.py [hunk 2]
```diff
@@ -446,6 +446,10 @@ def all_endtoend_scenarios(test_object):
         "TRACING_CONFIG_NONDEFAULT",
         weblog_env={
             "DD_TRACE_HTTP_SERVER_ERROR_STATUSES": "200-201,202",
+            "DD_TRACE_CLIENT_IP_ENABLED": "true",
+            "DD_TRACE_CLIENT_IP_HEADER": "custom-ip-header",
+            # disable ASM to test non asm client ip tagging
```
Comment (mabdinur) [inline utils/_context/_scenarios/__init__.py:451]:
```suggestion
            # disable ASM to test http.client _p collection via tracing integrations
```

File: manifests/php.yml [hunk 3]
```diff
@@ -316,6 +316,7 @@ tests/:
     test_miscs.py:
       Test_Miscs: missing_feature
   test_config_consistency.py:
+    Test_Config_ClientIPHeader_Configured: v1.3,0
```
Comment (cbeauchesne) [inline manifests/php.yml:319]:
```suggestion
    Test_Config_ClientIPHeader_Configured: v1.3.0
```

**Review by cbeauchesne — CHANGES_REQUESTED** [review]
Just a small typo.

**Review by cbeauchesne — APPROVED** [review]

File: manifests/nodejs.yml [hunk 4]
```diff
@@ -560,6 +560,8 @@ tests/:
       Test_RemoteConfigurationUpdateSequenceLiveDebugging: *ref_5_16_0 #actual version unknown
       Test_RemoteConfigurationUpdateSequenceLiveDebuggingNoCache: irrelevant (cache is implemented)
   test_config_consistency.py:
+    Test_Config_ClientIPHeader_Configured: *ref_5_15_0
```
Comment (simon-id) [inline manifests/nodejs.yml:563]: What is this version number coming from ?

Comment (mabdinur) [inline manifests/nodejs.yml:563]:
```suggestion    Test_Config_ClientIPHeader_Configured: v5.22.0
```

File: manifests/nodejs.yml [hunk 5]
```diff
@@ -560,6 +560,7 @@ tests/:
       Test_RemoteConfigurationUpdateSequenceLiveDebugging: *ref_5_16_0 #actual version unknown
       Test_RemoteConfigurationUpdateSequenceLiveDebuggingNoCache: irrelevant (cache is implemented)
   test_config_consistency.py:
+    Test_Config_ClientIPHeader_Precedence: missing_feature (all headers listed in the RFC are not supported)
```
Comment (mabdinur) [inline manifests/nodejs.yml:563]:
```suggestion
    Test_Config_ClientIPHeader_Configured: v5.22.0
    Test_Config_ClientIPHeader_Precedence: missing_feature (all headers listed in the RFC are not supported)
```

**Comment by mabdinur** [comment]
CI failures are unreleated (ex: due to mysql issues)

File: manifests/nodejs.yml (hunk 4 above)
Comment (simon-id) [reply manifests/nodejs.yml]: Please don't do that, I've been working hard to not do that for years

Comment (simon-id) [reply manifests/nodejs.yml]: Either leave it missing_feature, or ask us to find the correct version

**Pull request merged**